daemon/stores/workspaces.py

```diff
@@ -161,26 +165,25 @@ def bob_the_builder(self, id: DaemonID, files: List[UploadFile]):
         self._handle_files(workspace_id=id, files=files)
         daemon_file = DaemonFile(workdir=workdir)
         network_id = Dockerizer.network(workspace_id=id)
-        image_id = Dockerizer.build(
-            workspace_id=id, daemon_file=daemon_file
-        )
+        image_id = Dockerizer.build(workspace_id=id, daemon_file=daemon_file)
         _min, _max = random_port_range()
         return workdir, daemon_file, network_id, image_id, _min, _max
 
-    def _set_id(self, id, files, workdir, daemon_file, network_id,
-                image_id, _min, _max):
+    def _set_id(
+        self, id, files, workdir, daemon_file, network_id, image_id, _min, _max
+    ):
         # Move this to the Consumer thread
         if id in self:
             if files:
-                self[id].arguments.files.extend(
-                    [f.filename for f in files]
-                )
+                self[id].arguments.files.extend([f.filename for f in files])
             self[id].arguments.jinad = {
                 'build': daemon_file.build,
                 'dockerfile': daemon_file.dockerfile,
             }
             self[id].metadata.image_id = image_id
-            self[id].arguments.requirements = daemon_file.requirements
+            self[id].arguments.requirements = list(
```

Comment:
Please handle `requirements` in `DaemonFile` property

Reply:
`requirements` is one of the build-args in devel.dockerfile. We need to pass the args as space-separated so that docker build-args can understand it (it doesn't understand a python list). Hence the fix should be in the pydantic model.